mm/kfence/kfence-test.c

```diff
@@ -345,16 +345,20 @@ static void test_shrink_memcache(struct kunit *test)
  */
 static void test_free_bulk(struct kunit *test)
 {
-	const size_t size = 1 + prandom_u32_max(300);
-	void *objects[4];
+	int iter;
 
-	objects[0] = test_alloc(test, size, GFP_KERNEL, ALLOCATE_ANY);
-	objects[1] = kmalloc(size, GFP_KERNEL);
-	objects[2] = test_alloc(test, size, GFP_KERNEL, ALLOCATE_ANY);
-	objects[3] = kmalloc(size, GFP_KERNEL);
-	kfree_bulk(ARRAY_SIZE(objects), objects);
+	for (iter = 0; iter < 10; ++iter) {
+		const size_t size = 1 + prandom_u32_max(300);
+		void *objects[4];
 
-	KUNIT_EXPECT_FALSE(test, report_available());
+		objects[0] = test_alloc(test, size, GFP_KERNEL, ALLOCATE_RIGHT);
```

Comment:
OOC, did you mean to change ALLOCATE_ANY to ALLOCATE_RIGHT/LEFT?

Reply:
Yes, I wanted to test both LEFT / RIGHT explicitly because I discovered some rare issue and didn't want to leave this up to randomness.

This makes the test more deterministic, and we do explicitly test both.

I think this was related to SLUB writing into the freeptr and the code that was trying to fix up the canary values. In the end I fixed it by making it never write into the redzone (somewhere in the following commits).

Reply:
Cool, thanks!